### PR TITLE
Improve gulp-if + add gulp-uglify

### PR DESCRIPTION
**Pull request by tkrotoff**

### CODE REVIEW

**Comment by vvakame** [comment]
*gulp-if/gulp-if.d.ts*

@pspeter3 could you review this PR?

---

*gulp-uglify/gulp-uglify.d.ts*

check list

* [ ] is collect [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)?
  * https://www.npmjs.com/package/gulp-uglify
  * http://bower.io/search/?q=gulp-uglify
  * others?
* [X] has a [test file](http://definitelytyped.org/guides/contributing.html#tests)? (gulp-uglify/gulp-uglify-tests.ts or others)
* [ ] pass the Travic-CI test?

File: gulp-uglify/gulp-uglify.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Type definitions for gulp-uglify v1.2.0
+// Project: https://github.com/terinjokes/gulp-uglify
+// Definitions by: Tanguy Krotoff <https://github.com/tkrotoff>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+/// <reference path="../node/node.d.ts" />
+
+declare module 'gulp-uglify' {
+    function uglify(opts?: any): NodeJS.ReadWriteStream;
```
Comment (pspeter3) [inline gulp-uglify/gulp-uglify.d.ts]: Is there an interface for the opts?

**Comment by pspeter3** [comment]
Looks good but it would be nice if `uglify` had an actual options interface

**Comment by vvakame** [comment]
@pspeter3 thanks!
@tkrotoff please check his comment and resolve conflict.

**Comment by tkrotoff** [comment]
@pspeter3 gulp-uglify has a list of options [here](https://github.com/terinjokes/gulp-uglify#options) + you can pass any option from UglifyJS2: [1](https://github.com/mishoo/UglifyJS2#the-simple-way), [2](http://lisperator.net/uglifyjs/codegen), [3](http://lisperator.net/uglifyjs/compress). In the end I've found the list of possible options to be large and complex so I went for the easy path => `opts?: any`.
I believe if one finds value in listing all possible options, one will implement it.

**Comment by pspeter3** [comment]
@tkrotoff If you feel like that is the best user experience, then sure. But the type definition is often helpful as documentation for the package.

**Comment by tkrotoff** [comment]
@pspeter3 don't get me wrong, of course it would be useful: in a perfect world everything should be defined. Still a lot of packages use the same approach: [1](https://github.com/borisyankov/DefinitelyTyped/search?q=%22options%3F%3A+any%22&type=Code), [2](https://github.com/borisyankov/DefinitelyTyped/search?q=%22opts%3F%3A+any%22&type=Code).
- Even with `opts?: any` this "package" is useful
- `opts?: any` is not wrong, just less precise
- It can be changed in the future
- Most won't customize the uglify output
- It is for gulpfile.ts, something less important than your "real" application code
- Lots of work: lots of options + you probably want to create definitions for UglifyJS2 itself since gulp-uglify inherits from its options)

I did specify `opts` interface in other packages when the amount of work were less important: [1](https://github.com/borisyankov/DefinitelyTyped/blob/879ba33f4d8bbe4b2651fb116035fc8df0003c90/gulp-minify-html/gulp-minify-html.d.ts#L9), [2](https://github.com/borisyankov/DefinitelyTyped/blob/eef34fb148ff351f3dcd893ed3890e4613f4d649/gulp-debug/gulp-debug.d.ts#L9), [3](https://github.com/borisyankov/DefinitelyTyped/blob/eef34fb148ff351f3dcd893ed3890e4613f4d649/gulp-protractor/gulp-protractor.d.ts#L10), [4](https://github.com/borisyankov/DefinitelyTyped/blob/eef34fb148ff351f3dcd893ed3890e4613f4d649/gulp-rev-replace/gulp-rev-replace.d.ts#L9), [5](https://github.com/borisyankov/DefinitelyTyped/blob/eef34fb148ff351f3dcd893ed3890e4613f4d649/gulp-rev/gulp-rev.d.ts#L9), [6](https://github.com/borisyankov/DefinitelyTyped/blob/eef34fb148ff351f3dcd893ed3890e4613f4d649/gulp-size/gulp-size.d.ts#L9), [7](https://github.com/borisyankov/DefinitelyTyped/blob/eef34fb148ff351f3dcd893ed3890e4613f4d649/gulp-useref/gulp-useref.d.ts#L9), [8](https://github.com/borisyankov/DefinitelyTyped/blob/eef34fb148ff351f3dcd893ed3890e4613f4d649/gulp-watch/gulp-watch.d.ts#L9), [9](https://github.com/borisyankov/DefinitelyTyped/blob/eef34fb148ff351f3dcd893ed3890e4613f4d649/merge2/merge2.d.ts#L9).

**Comment by pspeter3** [comment]
Sounds good. Thanks for the definitions! :smile:

**Comment by tkrotoff** [comment]
Ping @vvakame

**Comment by vvakame** [comment]
@tkrotoff please resolve conflict.

**Comment by tkrotoff** [comment]
Another PR that added gulp-uglify was merged: #4902 (Jul 14)
Another PR that fixed gulp-if condition parameter was merged: #5455 (Aug 21)

**Pull request closed without merging**